### ISSUE TITLE
Version 1.10 does not conpile on centos7's old gcc (4.8.5).

### ISSUE THREAD

**Issue by Simon-Janos** [gabime/spdlog#2448]
Version 1.10 does not conpile on centos7's old gcc (4.8.5).

Luckily only a small fix can cure it:

```
bash-4.2# git diff
--- a/include/spdlog/pattern_formatter-inl.h
+++ b/include/spdlog/pattern_formatter-inl.h
@@ -1053,7 +1053,7 @@ SPDLOG_INLINE std::unique_ptr<formatter> pattern_formatter::clone() const
     }
     auto cloned = details::make_unique<pattern_formatter>(pattern_, pattern_time_type_, eol_, std::move(cloned_custom_formatters));
     cloned->need_localtime(need_localtime_);
-    return cloned;
+    return std::unique_ptr<formatter>(std::move(cloned));
 }

 SPDLOG_INLINE void pattern_formatter::format(const details::log_msg &msg, memory_buf_t &dest)
bash-4.2#
```

**Comment by gabime** [comment]
Duplicate of #2427 
A pr with a special ifdef for old gcc would be welcome.

**Comment by Simon-Janos** [comment]
No need for ifdef as explicit casting is always better.

**Comment by gabime** [comment]
🪩 Not sure why the cloned, which is already unique ptr copied to a new one and returned.

**Comment by Simon-Janos** [comment]
It is basically casted back to its base class as the return type suggests. The original code looks cpp implementation dependent, this  version tells the compiler what to do. I expect the original code to do something similar anyway.

**Comment by gabime** [comment]
What is the compiler error? If this is indeed dup of #2427 you can just return std::move(cloned) to make the old compiler happy (but not the newer ones).
So I suggest to ifdef this and return the move since I dont want affect newer compilers again.

**Comment by gabime** [comment]
Fixed in pr #2449
